src/lib/route-helpers.js

```diff
@@ -14,30 +14,30 @@ const promiseResponder = (promise, Component) => (req, res, next) => promise(req
 
 const dateChecker = (d) => !!d && moment.tz(d, dateFormat, true, 'Europe/London').format();
 
-const home = (query, ErrorReporter) => new Promise((resolve) => {
-  const fromDate = dateChecker(query && query.from);
-  if (ErrorReporter && fromDate === 'Invalid date') {
-    throw new ErrorReporter('Must provide "from" date parameter, and optionally a "to" date');
+const dateError = RequestError => (value, message) => {
+  const date = dateChecker(value);
+  if (date === 'Invalid date') {
+    throw new RequestError(message);
   }
-  const toDate = dateChecker(query && query.to);
-  if (ErrorReporter && toDate === 'Invalid date') {
-    throw new ErrorReporter(`Make sure the date format is "${dateFormat}" (time is ignored)`);
-  }
-  const searchGroup = query && query.currentGroup;
+  return date;
+};
 
+const home = (validate = dateChecker) => (query = {}) => new Promise((resolve) => {
+  const fromDate = validate(query.from, 'Must provide "from" date parameter, and optionally a "to" date');
+  const searchGroup = query.currentGroup;
   return resolve(model(
     (fromDate ? fromDate : moment.tz('Europe/London').startOf('month').format()),
-    toDate,
+    validate(query.to, `Make sure the date format is "${dateFormat}" (time is ignored)`),
```

Comment:
You've still got an uneasy mix of logic spread between the model call and declarations (further up), it's probably better to pick one style and stick to it.

Reply:
Fixed I think